packages/eas-build-job/src/metadata.ts

```diff
@@ -173,6 +183,8 @@ export const MetadataSchema = Joi.object({
   credentialsSource: Joi.string().valid('local', 'remote'),
   sdkVersion: Joi.string(),
   runtimeVersion: Joi.string(),
+  fingerprintGCSBucketKey: Joi.string(),
+  projectFingerprintUrl: Joi.string(),
```

Comment:
You would need to add 
```ts
export const FingerprintSourceSchema = Joi.object<FingerprintSource>({
  type: Joi.string()
    .valid(...Object.values(FingerprintSourceType))
    .required(),
})
  .when(Joi.object({ type: FingerprintSourceType.GCS }).unknown(), {
    then: Joi.object({
      type: Joi.string().valid(ArchiveSourceType.GCS).required(),
      bucketKey: Joi.string().required(),
    }),
  })
  .when(Joi.object({ type: ArchiveSourceType.PATH }).unknown(), {
    then: Joi.object({
      type: Joi.string().valid(ArchiveSourceType.PATH).required(),
      path: Joi.string().required(),
    }),
  });
```

And then do 
```suggestion
  fingerprintSource: FingerprintSourceSchema,
```